### PR TITLE
Making addPermittedAccount and removePermittedAccount methods idempotent

### LINKED ISSUE #15708: Make permissionController methods: addPermittedAccount and removePermittedAccount idempotent

### PR DESCRIPTION
**Pull request by jpuri**
Fixes #15708

Making addPermittedAccount and removePermittedAccount methods idempotent.

### CODE REVIEW

**Review by JurbinM — APPROVED** [review]

**Review by NiranjanaBinoy — DISMISSED** [review]
LGTM

File: app/scripts/controllers/permissions/background-api.test.js [hunk 1]
```diff
@@ -34,22 +34,17 @@ describe('permission background API methods', () => {
       );
     });
 
-    it('throws if the specified account is already permitted', () => {
+    it('does not call permissionController.updateCaveat if the specified account is already permitted', () => {
```
Comment (digiwand) [inline app/scripts/controllers/permissions/background-api.test.js:37]: nit: can we update this phrase to indicate that it's testing the opposite of the above test? To me, it is a little cryptic understanding the intention of `permissionController.updateCaveat` in this test

```suggestion
    it('does not add a permitted account', () => {
```

Comment (digiwand) [inline app/scripts/controllers/permissions/background-api.test.js:37]: nit: can we update this phrase to indicate that it's testing the opposite of the above test? To me, it is a little cryptic understanding the intention of `permissionController.updateCaveat` in this test

```suggestion
    it('does not add a permitted account if the specified account is already permitted', () => {
```

File: app/scripts/controllers/permissions/background-api.js [hunk 2]
```diff
@@ -14,9 +14,7 @@ export function getPermissionBackgroundApiMethods(permissionController) {
       );
 
       if (existing.value.includes(account)) {
-        throw new Error(
-          `eth_accounts permission for origin "${origin}" already permits account "${account}".`,
-        );
+        return;
```
[the remark this answers is not in the document]
Comment (jpuri) [reply app/scripts/controllers/permissions/background-api.js:17]: If request is called again for same account it should not throw error but return old account that was added.

This is to make the function calls idempotent.

[the remark this answers is not in the document]
Comment (digiwand) [reply app/scripts/controllers/permissions/background-api.js:17]: Thanks for the explanation. I didn't realize that throwing an error could cause a function to not be idempotent. I thought, if an error is thrown, given the same inputs, it should produce the same error. 

There's a good chance I'm missing context. I'd love to extend this talk, maybe over the next meeting we have together

**Review by digiwand — APPROVED** [review]

**Review by darkwing — APPROVED** [review]
This seems OK to me.  In retrospect, an error here seems a bit harsh, IMO.

**Pull request merged**